### ISSUE TITLE
Design question: Why Topic.subscribe does not mark subscribe as an effect?

### ISSUE THREAD

**Issue by qingwei91** [functional-streams-for-scala/fs2#1485]
Hi, can I know why the signature of subscribe is 

```scala
def subscribe(maxQueued: Int): Stream[F, A]

instead of 
def subscribe(maxQueued: Int): FStream[F, A]]
````

I would think the 2nd version makes more sense because `subscribe` (ie start listening) is an effect and pulling data from the subscription is a different effect. Wonder if I am missing anything?

https://github.com/functional-streams-for-scala/fs2/blob/series/1.0/core/shared/src/main/scala/fs2/concurrent/Topic.scala#L56

**Comment by SystemFw** [comment]
It's a legit question, see this PR for `subscribeAwait: Resource[F, Stream[F, A]]` (`F` is not enough because you want to handle unsubscribing)
https://github.com/functional-streams-for-scala/fs2/pull/1366

This is currently blocked by bincompat

**Comment by qingwei91** [comment]
I see, thanks, that makes more sense

> (Just for pedantry, subscribing and pulling are both effects, not side effects, but it's not really pertinent to your question :) )

You're right 👍 

Closing as this is already tracked
